### ISSUE TITLE
Error raised from DataSink on Python 3

### ISSUE THREAD

**Issue by mwaskom** [nipy/nipype#2091]
I'm transitioning some workflow code to Python3 and have run into an issue in nipype that I think is a Python3 conversion problem. Here is the traceback I see:

```
Traceback (most recent call last):
  File "/home/mwaskom/anaconda2/envs/py36/bin/run_fmri.py", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/mwaskom/code/lyman/scripts/run_fmri.py", line 525, in <module>
    main(sys.argv[1:])
  File "/home/mwaskom/code/lyman/scripts/run_fmri.py", line 355, in main
    lyman.run_workflow(ffx, "ffx", args)
  File "/home/mwaskom/code/lyman/lyman/frontend.py", line 196, in run_workflow
    wf.run(plugin, plugin_args)
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/pipeline/engine/workflows.py", line 590, in run
    runner.run(execgraph, updatehash=updatehash, config=self.config)
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/pipeline/plugins/base.py", line 266, in run
    result=result))
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/pipeline/plugins/base.py", line 306, in _clean_queue
    raise RuntimeError("".join(result['traceback']))
RuntimeError: Traceback (most recent call last):
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/pipeline/plugins/base.py", line 255, in run
    result=result))
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/pipeline/plugins/base.py", line 306, in _clean_queue
    raise RuntimeError("".join(result['traceback']))
RuntimeError: Traceback (most recent call last):
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/interfaces/io.py", line 72, in copytree
    os.makedirs(dst)
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/os.py", line 220, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/home/mwaskom/studies/fluid/analysis/nback-block_load/gf52/ffx/mni/smoothed/n0'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/pipeline/plugins/multiproc.py", line 52, in run_node
    result['result'] = node.run(updatehash=updatehash)
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/pipeline/engine/nodes.py", line 372, in run
    self._run_interface()
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/pipeline/engine/nodes.py", line 482, in _run_interface
    self._result = self._run_command(execute)
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/pipeline/engine/nodes.py", line 613, in _run_command
    result = self._interface.run()
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/interfaces/base.py", line 1082, in run
    outputs = self.aggregate_outputs(runtime)
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/interfaces/base.py", line 1153, in aggregate_outputs
    predicted_outputs = self._list_outputs()
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/interfaces/io.py", line 757, in _list_outputs
    copytree(src, dst)
  File "/home/mwaskom/anaconda2/envs/py36/lib/python3.6/site-packages/nipype/interfaces/io.py", line 74, in copytree
    if 'File exists' in why:
TypeError: argument of type 'FileExistsError' is not iterable
Interface DataSink failed to run. 
```

The issue is [here](https://github.com/nipy/nipype/blob/master/nipype/interfaces/io.py#L74). I believe the problem is that Exception objects no longer duck-type as strings in Python 3. In other words:

```
"error" in ValueError("This is an error")
```

returns `True` on Python 2 and raises `TypeError` on Python 3.

I guess the line can be changed to `if 'FileExists' in str(why)`, but I'm not certain.

### Platform details:

```
$ python -c "import nipype; print(nipype.get_info()); print(nipype.__version__)"
{'nibabel_version': '2.1.0', 'sys_executable': '/home/mwaskom/anaconda2/bin/python', 'networkx_version': '1.11', 'numpy_version': '1.11.2', 'sys_platform': 'linux2', 'sys_version': '2.7.12 |Anaconda custom (64-bit)| (default, Jul  2 2016, 17:42:40) \n[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]', 'commit_source': u'installation', 'commit_hash': '9ac1058', 'pkg_path': '/home/mwaskom/anaconda2/lib/python2.7/site-packages/nipype', 'nipype_version': u'1.0.0-dev', 'traits_version': '4.6.0', 'scipy_version': '0.18.1'}
1.0.0-dev
```

**Comment by mwaskom** [comment]
See #2092, which I found fixed this issue for my case. I'm not sure what a clever way is to search for this pattern elsewhere in nipype, but I found a couple of other instances in the `io.py` module.

**Comment by mwaskom** [comment]
Closed with #2092
